install/gcp/upgrade_tools/db_migrator.py

```diff
@@ -63,62 +84,92 @@ def drop_column(table, column):
     tables = base.metadata.tables
 
     schema_update_actions_method = 'get_schema_update_actions'
 
-    for subclass in base_subclasses:
-        get_schema_update_actions = getattr(subclass,
+    for dao_class in dao_classes:
+        get_schema_update_actions = getattr(dao_class,
                                             schema_update_actions_method,
                                             None)
         if (not callable(get_schema_update_actions) or
-                subclass.__tablename__ not in tables):
+                dao_class.__tablename__ not in tables):
             LOGGER.warn('Method: %s is not callable or Table: %s doesn\t '
                         'exist', schema_update_actions_method,
-                        subclass.__tablename__)
+                        dao_class.__tablename__)
             continue
-        LOGGER.info('Updating table %s', subclass.__tablename__)
+        LOGGER.info('Updating table %s', dao_class.__tablename__)
         # schema_update will require the Table object.
-        table = tables.get(subclass.__tablename__)
+        table = tables.get(dao_class.__tablename__)
         schema_update_actions = get_schema_update_actions()
         for column_action, columns in schema_update_actions.iteritems():
-            column_action = column_action.upper()
-            if column_action in COLUMN_ACTION_MAPPING:
-                for column in columns:
-                    try:
-                        COLUMN_ACTION_MAPPING.get(column_action)(table,
-                                                                 column)
-                    except OperationalError:
-                        LOGGER.info('Failed to update db schema, table=%s',
-                                    subclass.__tablename__)
-                    except Exception:  # pylint: disable=broad-except
-                        LOGGER.exception(
-                            'Unexpected error happened when attempting '
-                            'to update database schema, table: %s',
-                            subclass.__tablename__)
+            if column_action in [ColumnAction.CREATE, ColumnAction.DROP]:
+                _create_or_drop_columns(column_action, columns, table)
+            elif column_action in [ColumnAction.ALTER]:
+                _alter_columns(column_action, columns, table)
             else:
-                LOGGER.warn('Columns: %s, ColumnAction %s doesn\'t '
-                            'exist.', columns, column_action)
+                LOGGER.warn('Unknown column action: %s', column_action)
+
+
+def _alter_columns(column_action, columns, table):
+    """Alter columns.
+
+    Args:
+        column_action (str): Column Action.
+        columns (dict): A dictionary of old_column: new_column.
+        table (sqlalchemy.schema.Table): The sql alchemy table object.
+    """
+    column_action = column_action.upper()
+    for old_column, new_column in columns.iteritems():
```

Comment:
Is there any check to make sure the alter command doesn't run all the time?  If there are data in the tables, the alter command might take a while.

Reply:
The same ALTER command will be executed but mysql will detect if there is any changes before actually applying it